src/views/Update/UpdatePage/index.js

```diff
@@ -0,0 +1,257 @@
+          setSaving(false);
+        });
+      }
+    };
+
+    const saveButton = isSaving
+      ? <GordonLoader size={32} />
+      : <Button
+          variant="contained"
+          color="secondary"
+          onClick={handleSaveButtonClick}
+          disabled={hasNoChange}
+        >
+          Update
+        </Button>;
+
+    const personalInfo = [
+      {label: "Salutation", name: "salutation", value: userInfo.salutation},
+      {label: "First Name", name: "firstname", value: userInfo.firstname},
+      {label: "Last Name", name: "lastname", value: userInfo.lastname},
+      {label: "Middle Name", name: "middlename", value: userInfo.middlename},
+      {label: "Preferred Name", name: "preferredname", value: userInfo.preferredname},
+      {label: "Married", name: "maritalstatus", value: userInfo.maritalstatus}
+    ];
+    const emailInfo = [
+      {label: "Personal Email", name: "personalemail", value: userInfo.personalemail},
+      {label: "Work Email", name: "workemail", value: userInfo.workemail},
+      {label: "Alternate Email", name: "alt_email", value: userInfo.alt_email},
+      {label: "Preferred Email", name: "preferredemail", value: userInfo.preferredemail, menuItem:
+        [{value: "Personal Email"}, {value: "Work Email"}, {value: "Alternate Email"}]
+      }
+    ];
+    const phoneInfo = [
+      {label: "Home Phone", name: "homephone", value: userInfo.homephone},
+      {label: "Work Phone", name: "workphone", value: userInfo.workphone},
+      {label: "Mobile Phone", name: "mobilephone", value: userInfo.mobilephone},
+      {label: "Preferred Phone", name: "preferredphone", value: userInfo.preferredphone, menuItem:
+        [{value: "Home Phone"}, {value: "Work Phone"}, {value: "Mobile Phone"}]
+      }
+    ];
+    const mailingInfo = [
+      {label: "Street", name: "address", value: userInfo.address},
+      {label: "City", name: "city", value: userInfo.city},
+      {label: "State", name: "state", value: userInfo.state},
+      {label: "Zip", name: "zip", value: userInfo.zip},
+      {label: "Country", name: "country", value: userInfo.country}
+    ];
+    const shouldContactForm = [
+      {label: "Do Not Contact", name: "doNotContact", checked: userInfo.doNotContact},
+      {label: "Do Not Mail", name: "doNotMail", checked: userInfo.doNotMail}
+    ];
+
+    const infoMap = (data) => {
+      return data.map((info) => {
+        let content =
+          typeof info.checked === "boolean"
+          ? <UpdateForm
+              info={info}
+              onChange={handleChange}
+            />
+          : typeof info.menuItem === "object"
+            ? <UpdateSelect
+                info={info}
+                onChange={handleChange}
+              />
+            : <UpdateGrid
+                info={info}
+                onChange={handleChange}
+              />
+        return content;
+      })
+    }
```

Comment:
You can simplify this a fair bit by changing the format of `data` slightly.

Firstly, notice that you're passing exactly the same props to `UpdateForm`, `UpdateSelect`, and `UpdateGrid`, which indicates that you can extract the logic for deciding which type of input this is to a custom component, e.g. `ProfileUpdateField`. Then, the content of `infoMap` would be just `data.map((info) => <ProfileUpdateField info={info} onChange={handleChange} />

Secondly, we can extract the `personalInfo`, `emailInfo`, etc. arrays out of the component so that they aren't recreated on every render. The only reason they're inside the component currently is because of the reference to `userInfo` for the `value` field. But notice that `value` is ALWAYS `userInfo[name]`. So we can just remove value and add it in when we map the info to components.

Thirdly, if `data` is an array of `{label: string, name: string, type: 'checkbox' | 'textField' | 'select', menuItems?: string[]}` objects, then we can greatly simplify the logic for deciding which input this is.

Putting together all three of those suggestions, you end up with for example:

```jsx
const infoMap = (fields) => {
  return fields.map((field) => <ProfileUpdateField label={field.label} name={field.name} value={userInfo[field.name]} type={label.type} menuItems={field.menuItems} onChange={handleChange} />
}

// In `src/views/Update/components/ProfileUpdateField/index.js`
const ProfileUpdateField = ({label, name, value, type, menuItems, onChange}) => {
  let field;
  switch (type) {
    case 'textField':
      field = (<TextField
        className="disable_select"
        style={{ width: 252, }}
        label={label}
        name={name}
        value={value}
        onChange={onChange}
      />);
      break;
    case 'checkbox':
      field = (<FormControlLabel
        control={<Checkbox checked={value}
        onChange={onChange} />}
        label={label}
        name={name}
      />);
      break;
    case 'select':
      field = (<FormControl style={{ width: 252 }}>
        <InputLabel>{label}</InputLabel>
        <Select
          label={label}
          name={name}
          value={value}
          onChange={onChange}
        >
          {menuItems.map((info) => (
            <MenuItem value={info.value}>{info.value}</MenuItem>
          ))}
        </Select>
      </FormControl>);
      break;
  }
  return (
    <Grid item xs={9} md={3} lg={3}>
      {content}
    </Grid>
  );
}
```

Since `infoMap` is now just a call to `map`, all it does is conceal what's happening without actually encapsulating anything. So I think you could just replace calls to `infoMap` with e.g. `personalInfoFields.map(...)` etc.